propagation.go

```diff
@@ -147,21 +146,14 @@ type HTTPHeadersCarrier http.Header
 // Set conforms to the TextMapWriter interface.
 func (c HTTPHeadersCarrier) Set(key, val string) {
```

Comment:
github won't let me comment on line 144, but I think we ought to include a quick example of how `HTTPHeadersCarrier` should be used in practice; or, alternatively, point to the examples in `Tracer.Inject` and `Tracer.Extract`.